utils/check_sources.py

```diff
@@ -1,259 +0,0 @@
-            yield lno + 1, "line too long"
-        if line.strip().startswith('#'):
-            continue
-        # m = not_ix_re.search(line)
-        # if m:
-        #     yield lno+1, '"' + m.group() + '"'
-        if is_const_re.search(line):
-            yield lno + 1, 'using == None/True/False'
-
-
-@checker('.py', only_pkg=True)
-def check_fileheader(fn, lines):
-    # line number correction
-    c = 1
-    if lines[0:1] == ['#!/usr/bin/env python\n']:
-        lines = lines[1:]
-        c = 2
-
-    llist = []
-    docopen = False
-    for lno, l in enumerate(lines):
-        llist.append(l)
-        if lno == 0:
-            if l != '# -*- coding: utf-8 -*-\n':
-                yield 1, "missing coding declaration"
-        elif lno == 1:
-            if l != '"""\n' and l != 'r"""\n':
-                yield 2, 'missing docstring begin (""")'
-            else:
-                docopen = True
-        elif docopen:
-            if l == '"""\n':
-                # end of docstring
-                if lno <= 4:
-                    yield lno + c, "missing module name in docstring"
-                break
-
-            if l != '\n' and l[:4] != '    ' and docopen:
-                yield lno + c, "missing correct docstring indentation"
-
-            if lno == 2:
-                # if not in package, don't check the module name
-                modname = fn[:-3].replace('/', '.').replace('.__init__', '')
-                while modname:
-                    if l.lower()[4:-1] == modname:
-                        break
-                    modname = '.'.join(modname.split('.')[1:])
-                else:
-                    yield 3, "wrong module name in docstring heading"
-                modnamelen = len(l.strip())
-            elif lno == 3:
-                if l.strip() != modnamelen * '~':
-                    yield 4, "wrong module name underline, should be ~~~...~"
-
-    else:
-        yield 0, "missing end and/or start of docstring..."
-
-    # check for copyright and license fields
-    license = llist[-2:-1]
-    if not license or not license_re.match(license[0]):
-        yield 0, "no correct license info"
-
-    ci = -3
-    copyright = llist[ci:ci + 1]
-    while copyright and copyright_2_re.match(copyright[0]):
-        ci -= 1
-        copyright = llist[ci:ci + 1]
-    if not copyright or not copyright_re.match(copyright[0]):
-        yield 0, "no correct copyright info"
-
-
-@checker('.py', '.html', '.rst')
-def check_whitespace_and_spelling(fn, lines):
-    for lno, line in enumerate(lines):
-        if '\t' in line:
-            yield lno + 1, "OMG TABS!!!1 "
-        if line[:-1].rstrip(' \t') != line[:-1]:
-            yield lno + 1, "trailing whitespace"
-        for word in misspellings:
-            if word in line and 'ALLOW-MISSPELLING' not in line:
-                yield lno + 1, '"%s" used' % word
-
-
-bad_tags = ['<u>', '<s>', '<strike>', '<center>', '<font']
-
-
-@checker('.html')
-def check_xhtml(fn, lines):
```

Comment:
It seems this test has been lost. Is this intended? I'm okay to remove this, but please let us know such important change.

Reply:
Did you remove any other?

Reply:
> It seems this test has been lost. Is this intended? I'm okay to remove this, but please let us know such important change.

Yes, I removed others but they were useless. I called this out in [the commit message](https://github.com/sphinx-doc/sphinx/pull/4118/commits/672e7c444fd484a293e624b1ef41e975f4960e5c). Is this adequate?

Reply:
Okay, I understand. Sorry, I did not read each commit messages.
+1 for dropping them.

It would be nice if we introduce HTML lints to our tests instead (in future).